include/istio/utils/attributes_builder.h

```diff
@@ -89,18 +89,20 @@ class AttributesBuilder {
     }
   }
 
-  void AddProtobufStringMap(
-      const std::string& key,
-      const google::protobuf::Map<std::string, ::std::string>& string_map) {
-    if (string_map.empty()) {
+  void AddProtoStructStringMap(const std::string& key,
+                               const google::protobuf::Struct& struct_map) {
+    if (struct_map.fields().empty()) {
       return;
     }
     auto entries = (*attributes_->mutable_attributes())[key]
                        .mutable_string_map_value()
                        ->mutable_entries();
     entries->clear();
-    for (const auto& map_it : string_map) {
-      (*entries)[map_it.first] = map_it.second;
+    for (const auto& field : struct_map.fields()) {
+      // Ignore all fields that are not string.
+      if (!field.second.string_value().empty()) {
```

Comment:
Is it okay to use `.empty()` to check if a field is string? what if a string attribute that is empty?

Reply:
Then it's not worth to send it over, I think :). Anyway, change to kind_case.